backend/api/blueprints/leads.py

```diff
@@ -222,6 +222,11 @@ def _create_new_lead(request, valid_data_sources=VALID_DATA_SOURCES):
         if field != "id" and field in DEFAULT_LEAD_FIELDS and value is not None
     }
 
+    # validate company name field
+    company_name = body.get("company_name")
+    if not company_name:
```

Comment:
We can probably get this to handle strings that consist of only whitespace characters using `company_name.strip`. You might also have to make the default return from `body.get` and empty string `""` so `strip doesn't get called on `None`.